LBH.AdultSocialCare.Functions.Payruns/Services/PayrunGenerator.cs

```diff
@@ -16,22 +18,24 @@ public class PayrunGenerator
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly ICarePackageGateway _carePackageGateway;
         private readonly IPayrunGateway _payrunGateway;
+        private readonly ILogger<PayrunGenerator> _logger;
         private readonly InvoiceGenerator _invoiceGenerator;
 
         public PayrunGenerator(
-            IHttpContextAccessor httpContextAccessor, ICarePackageGateway carePackageGateway,
-            IFundedNursingCareGateway fncGateway, IPayrunGateway payrunGateway)
+            InvoiceGenerator invoiceGenerator, IHttpContextAccessor httpContextAccessor,
+            ICarePackageGateway carePackageGateway, IPayrunGateway payrunGateway, ILogger<PayrunGenerator> logger)
         {
             _httpContextAccessor = httpContextAccessor;
             _carePackageGateway = carePackageGateway;
             _payrunGateway = payrunGateway;
+            _logger = logger;
 
-            _invoiceGenerator = new InvoiceGenerator(_carePackageGateway, fncGateway);
+            _invoiceGenerator = invoiceGenerator;
         }
 
         public async Task GenerateAsync()
         {
-            // TODO: VK: Handle payrun types
+            // TODO: VK: Handle payrun ID from SQS, add reprocessing for hang-up draft / in-progress payruns
             var draftPayruns = await _payrunGateway.GetDraftPayrunsAsync();
 
             foreach (var payrun in draftPayruns)
```

Comment:
payrun status should be changed to "in-progress" before starting invoice generation.

Reply:
sure, let's implement it in the end, since it more convenient to debug without need to constantly switch payrun back to draft :)